### DISCUSSION TITLE
Advice on ensuring the Windows images complete successfully?

### DISCUSSION THREAD

**Discussion by DamienDennehy** [community/community#160671] [Actions] [answered]
I've had some bad luck this week where I'm refreshing our runners internally, and I've had three or four failures in a couple of days. As the Windows image can take about 5 hours to install everything this means a lot of time is wasted. It's nothing specific, it was network traffic blips, a website being down, etc. I honestly think it was just one of those weeks.

Are there any best practices that I could follow to prevent this from happening?
Do I run 2 or 3 builds say 30 minutes apart so that if one fails I have another shot 30 minutes as opposed to waiting another 5 hours?

![image](https://user-images.githubusercontent.com/3924814/207685440-bcb3bb73-57a8-40a3-8e46-d193e18c0ada.png)

![image](https://user-images.githubusercontent.com/3924814/207685518-96e09dd8-c49e-4147-8cd8-07f1a26987b1.png)

Thanks!

EDIT: The MySql issue seems to be a legit issue at least.
EDIT2: Completed after several attempts. VSIX, MySql, WDK, all failed randomly.

**Comment by markatky46** [comment]
I feel your pain. We have the same thing happening here 9/10 builds

**Comment by wojsmol** [comment]
Please update to latest `main` branch where this is fixed https://github.com/actions/runner-images/pull/6769/

**Reply by markatky46** [reply]
I use main branch. That error ewas just 1 example. Right now i have errors with Postgresql which is another known issue. The ask here is to try and ensure that images still complete even if all the tooling does not get installed and then report what failed so we can make a decision on whether to publish or not.

**Comment by mikhailkoliada** [comment, marked as answer]
@DamienDennehy Hello! As we are using upstream sources to install the software there is no way to make it more safe - if the software source is unavailable it must be changed/fixed/etc, but nothing else can be done to make your builds green if a problem popped up.

**Reply by fajterini** [reply]
Maybe more conservative approach for version upgrade is needed. Just in month I see Two breaking bugs related to application upgrades. 
Also monitoring of VM build  success rate is needed. If someone is OK if build rate for last 30 days is 30%, something is not OK. 
https://github.visualstudio.com/runner-images/_pipeline/analytics/stageawareoutcome?definitionId=97&contextType=build

**Comment by rybal06** [comment]
I am curious, how does the MS team test image builds? 

Would it be possible to schedule a daily build out of this repo using a pipeline in which customers can view the results?

This would help in a few ways:
1. Identity whether a build issue is something transient or something others are experiencing.
2. Let us know whether or not we need to report the issue or whether MS is aware of the issue already.
3. Tell us if the issue is reproducible outside of our automation/CICD environment.

We see about an 88% failure rate on weekly builds over the past few months on the windows build. The Ubuntu build does not seem to be nearly as problematic.

**Comment by fajterini** [comment]
I have the same question :(  
I am watching success rate on their pipelines https://github.visualstudio.com/runner-images/_build and it's really bad :( 
Currently for me it's failing on Windows 11 SDK  Failed Compenent.Unreal.Android

**Comment by brgsstm** [comment]
I've been trying to successfully build the Windows images from the main branch now since that start of December, I estimate across ~50 attempts.  So far I have had 1 successful build.  That image however fails to provision when referencing in a VMSS (that's not the issue right now though).  Are the Windows profiles actually tested at merge??  Ubuntu is consistently successful.  My 1 successful build took 14 hours, Ubuntu averages 1hr 40mins.

**Reply by fajterini** [reply]
I have created issue for az module failed to install https://github.com/actions/runner-images/issues/6896 
Postgres sql tool where fixed at end of year 2022. 
I have now also issue with Visual studio Compenent.Unreal.Android ( I will create issue for this as well). On what step it failing for you ?? I am using main repo as source. I think we will need to be more pro active. 
I am started to use this image builder solution in December 2022, and I see only one issue after another :(

**Comment by brgsstm** [comment]
Another build, another failure...

    vhd: Package downloaded successfully in 2.24 seconds
    vhd: Starting Install Microsoft.DataTools.AnalysisServices.vsix...
    vhd: Unsuccessful exit code returned by the installation process: -532462766.
==> vhd: Provisioning step had errors: Running the cleanup provisioner, if present...

**Reply by fajterini** [reply]
This is for Windows 2022 ? Because i don't have this issue ? This is Visual studio ?

**Reply by brgsstm** [reply]
> This is for Windows 2022 ? Because i don't have this issue ? This is Visual studio ?

yes, windows2022 - see below for another failure (just happened as typing this)

**Comment by brgsstm** [comment]
and another...    

vhd: Starting Install wdksetup.exe...
    vhd: Installation successful in 99.96 seconds
    vhd: Starting Install WDK.vsix...
    vhd: Unsuccessful exit code returned by the installation process: -532462766.
==> vhd: Provisioning step had errors: Running the cleanup provisioner, if present...

**Reply by fajterini** [reply]
Thanks for the issue report I was checking build in the morning and found same issue :(
